Update deprecated APIs for Deno v1.40 (#1234)
denoland/deno_lint: docs/rules/no_deprecated_deno_api.md

  // ...
}
```

- `Deno.fdatasync`
- `Deno.fdatasyncSync`

`Deno.fdatasync` and `Deno.fdatasyncSync` are deprecated in favor of

> `Deno.FsFile.datasync` and `Deno.FsFile.datasyncSync` respectively.

Reviewer: ```suggestion
`Deno.FsFile.flushData` and `Deno.FsFile.flushDataSync` respectively.
```

- `Deno.fsync`
- `Deno.fsyncSync`

> `Deno.fsync` and `Deno.fsyncSync` are deprecated in favor of `Deno.FsFile.sync`
> and `Deno.FsFile.syncSync` respectively.

Reviewer: ```suggestion
`Deno.fsync` and `Deno.fsyncSync` are deprecated in favor of `Deno.FsFile.flushAll`
and `Deno.FsFile.flushAllSync`, respectively.
```
Ref https://github.com/denoland/deno/pull/22085

Reviewer: Ditto for other parts relating to this API.

- `Deno.isatty`

> `Deno.isatty` was deprecated in favor of `Deno.stdin.isTerminal()`.

Reviewer: ```suggestion
`Deno.isatty` was deprecated in favor of `Deno.stdin.isTerminal()`, `Deno.stdout.isTerminal()` and `Deno.stderr.isTerminal()`.
```

### Invalid:

```typescript
Deno.isatty(Deno.stdin.rid);

> Deno.isatty(Deno.stdout.rid);

Reviewer: ```suggestion
Deno.isatty(Deno.stdout.rid);
Deno.isatty(Deno.stderr.rid);
```

```

### Valid:

```typescript
Deno.stdin.isTerminal();

> Deno.stdout.isTerminal();

Reviewer: ```suggestion
Deno.stdout.isTerminal();
Deno.stderr.isTerminal();
```

```

- `Deno.resources()`

Deno.resources() was deprecated. There are no replacements for this API.

- `Deno.metrics()`

Deno.metrics() was deprecated. There are no replacements for this API.

**HTTP server API**

> - `Deno.serveHttp`

Reviewer 2: Can you add examples for `Deno.close()`/Deno.write etc too?
